lib/orbital/ext/active_merchant/active_merchant.rb

```diff
@@ -60,32 +71,94 @@ def commit(order, message_type, trace_number=nil)
                      })
       end
 
+      def build_new_order_xml(action, money, parameters = {})
+        requires!(parameters, :order_id)
+        xml = xml_envelope
+        xml.tag! :Request do
+          xml.tag! :NewOrder do
+            add_xml_credentials(xml)
+            # EC - Ecommerce transaction
+            # RC - Recurring Payment transaction
+            # MO - Mail Order Telephone Order transaction
+            # IV - Interactive Voice Response
+            # IN - Interactive Voice Response
+            xml.tag! :IndustryType, parameters[:industry_type] || ECOMMERCE_TRANSACTION
+            # A  - Auth Only No Capture
+            # AC - Auth and Capture
+            # F  - Force Auth No Capture and no online authorization
+            # FR - Force Auth No Capture and no online authorization
+            # FC - Force Auth and Capture no online authorization
+            # R  - Refund and Capture no online authorization
+            xml.tag! :MessageType, action
+            add_bin_merchant_and_terminal(xml, parameters)
+
+            yield xml if block_given?
+
+            xml.tag! :OrderID, format_order_id(parameters[:order_id])
+            xml.tag! :Amount, amount(money)
+            xml.tag! :Comments, parameters[:comments] if parameters[:comments]
+
+            # Add additional card information for tokenized credit card that must be placed after the above three elements
+            if action == AUTH_ONLY or action == AUTH_AND_CAPTURE
```

Comment:
Nit: I try to stay away from `or` since its low precedence can be confusing (non-issue here though).